tests/test_transpiler_unroller.py

```diff
@@ -111,3 +111,11 @@ def test_unroller(natives_1q, natives_2q):
         translated_circuit,
         native_gates=natives_1q | natives_2q | NativeGates.RZ | NativeGates.Z,
     )
+
+
+def test_measurements_non_comp_basis():
+    unroller = Unroller(native_gates=NativeGates.default())
+    circuit = Circuit(1)
+    circuit.add(gates.M(0, basis=gates.X))
+    transpiled_circuit = unroller(circuit)
+    assert isinstance(transpiled_circuit.queue[2], gates.M)
```

Comment:
maybe test also that the basis of the measurement is still `X`

Reply:
After transpiling the measurement gate should be in the compuational basis. It is a task of the unroller to translate the gates used to rotate the state before the measurement. I added a check and explicited it with a comment.